workers/api-server/src/http/admin.rs

```diff
@@ -488,19 +483,18 @@ fn parse_include_fillable_from_query_params(params: &QueryParams) -> Result<bool
 /// & potentially the price of the base asset
 async fn get_fillable_amount_and_price(
     meta: &OrderMetadata,
+    wallet_id: &WalletIdentifier,
     state: &State,
     price_reporter_job_queue: &PriceReporterQueue,
 ) -> Result<(Amount, Price), ApiServerError> {
-    let wallet_id =
-        state.get_wallet_for_order(&meta.id).await?.ok_or(not_found(ERR_WALLET_NOT_FOUND))?;
-    let wallet = state.get_wallet(&wallet_id).await?.ok_or(not_found(ERR_WALLET_NOT_FOUND))?;
+    let mut wallet = state.get_wallet(wallet_id).await?.ok_or(not_found(ERR_WALLET_NOT_FOUND))?;
 
     // Get an up-to-date order & balance.
     // The order stored on the `OrderMetadata` does not have the order amount
     // updated to account for fills, which will lead to an incorrect calculation
     // of the fillable amount.
-    let order = wallet.get_order(&meta.id).ok_or(not_found(ERR_ORDER_NOT_FOUND))?;
-    let balance = wallet.get_balance_for_order(order).ok_or(not_found(ERR_BALANCE_NOT_FOUND))?;
+    let order = wallet.remove_order(&meta.id).ok_or(not_found(ERR_ORDER_NOT_FOUND))?;
```

Comment:
`remove_order` doesn't make much sense here, why are we doing this?

Reply:
just to get rid of an unnecessary clone

Reply:
Get order is appropriate here, this clone is not going to make a dent in performance and as a reader it's confusing why we're removing the order